game/src/main/java/org/apollo/game/release/r181/decoders/EventKeyboardDecoder.java

```diff
@@ -0,0 +1,29 @@
+package org.apollo.game.release.r181.decoders;
+
+import org.apollo.game.message.impl.decode.EventKeyboardMessage;
+import org.apollo.net.codec.game.DataTransformation;
+import org.apollo.net.codec.game.DataType;
+import org.apollo.net.codec.game.GamePacket;
+import org.apollo.net.codec.game.GamePacketReader;
+import org.apollo.net.release.MessageDecoder;
+
+import java.time.Duration;
+
+/**
+ * @author Khaled Abdeljaber
+ */
+public class EventKeyboardDecoder extends MessageDecoder<EventKeyboardMessage> {
+	@Override
+	public EventKeyboardMessage decode(GamePacket packet) {
+		GamePacketReader reader = new GamePacketReader(packet);
+
+		final var events = new EventKeyboardMessage.EventKeyboard[reader.getLength() / 4];
+		for (int index = 0; index < events.length; index++) {
+			var keyPressed = reader.getUnsigned(DataType.BYTE, DataTransformation.NEGATE);
+			var duration = reader.getUnsigned(DataType.TRI_BYTE);
+			events[index] = new EventKeyboardMessage.EventKeyboard((int) keyPressed, Duration.ofMillis(duration));
```

Comment:
This field doesn't need to be a duration. You can store the integer and expose an API that returns a `Duration` over it if needed.

Reply:
Okay it returns a long now.

Reply:
You're right, I've changed it.